osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -73,6 +76,14 @@ protected override void Update()
                 return;
             }
 
+            // Bandaid Fix for pre-break HP drain
+            // Checks if the first object is Hit. After that, checks if the difference between Current time and lastHitTime exceeds 600ms
+            if (firstObjectIsHit && (Time.Current - lastHitTime) > 600)
```

Comment:
No.

Reply:
Can you explain the input-output of this method ApplyBeatmap(IBeatmap beatmap)?

Reply:
It receives in a beatmap, and it extracts whatever it needs out of it. It does whatever it wants to do.

Reply:
Maybe calculate something like `List<System.Tuple<int,int>> NonDrainSections`?

Reply:
/// <summary>
        /// The minimum duration required for a break to have any effect.
        /// </summary>
        public const double MIN_BREAK_DURATION = 650;

Reply:
I think we should approach this problem by changing broken beatmap Break Starttimes because, we can't predict into the future and assure if the Current Hitobject that we're hitting is the closest Hitobject to the upcoming Breaktime.

Reply:
@mcendu's idea is probably what you need to solve this (although the tuple must be of `double,double` because that's the type we use to store time values and also must be a `private` local). 
where you want to calculate this would be inside the `ApplyBeatmap(beatmap)` method, now think on how to calculate it, search off what `Beatmap` has. (should be fairly simple)